src/gt4py/next/otf/compilation/build_systems/compiledb.py

```diff
@@ -227,29 +227,42 @@ def _cc_prototype_program_source(
     )
 
 
-def _cc_find_compiledb(
-    prototype_program_source: stages.ProgramSource, cache_lifetime: config.BuildCacheLifetime
-) -> Optional[pathlib.Path]:
+def _cc_get_compiledb(
+    renew_compiledb: bool,
+    prototype_program_source: stages.ProgramSource,
+    build_type: config.CMakeBuildType,
+    cmake_flags: list[str],
+    cache_lifetime: config.BuildCacheLifetime,
+) -> pathlib.Path:
     cache_path = cache.get_cache_folder(
         stages.CompilableSource(prototype_program_source, None), cache_lifetime
     )
+
+    with lock.Lock(str(cache_path / "compiledb.lock"), lifetime=120):  # type: ignore[attr-defined] # mypy not smart enough to understand custom export logic
+        if not renew_compiledb and (compiled_db := _cc_find_compiledb(cache_path=cache_path)):
+            return compiled_db
+        return _cc_create_compiledb(
+            cache_path=cache_path,
+            prototype_program_source=prototype_program_source,
+            build_type=build_type,
+            cmake_flags=cmake_flags,
+        )
+
+
+def _cc_find_compiledb(cache_path: pathlib.Path) -> Optional[pathlib.Path]:
     compile_db_path = cache_path / "compile_commands.json"
     if compile_db_path.exists():
         return compile_db_path
     return None
 
 
 def _cc_create_compiledb(
+    cache_path: pathlib.Path,
```

Comment:
```suggestion
    path: pathlib.Path,
```
I would argue this has changed it's meaning now.